mozilla/blurts-server
# MNTOR-1115: add sentry to v2

# References: 
Jira: MNTOR-1115
Figma: n/a




# Description
* Adding sentry and modifying errors (by utilizing extended classes) to control noises in Sentry & provide better feedback for the frontend
* New Sentry project created, moving the old Sentry DSN to its own env var (legacy site has own sentry DSN now)
* Env vars are updated on the deployment infra side
* Going forward, we should
  * Use `throw new Error` only if it's a 500+, these errors will be seen as server side issues by Sentry and recorded
  * Any responses with statusCode 300s and 400s (using a class in `/utils/errors.js`) will NOT be picked up by Sentry. These are usually errors that the frontend would want to see

# How to test
1) In `src/controllers/landing.js`, L12, add an error: 
```
  throw new Error('Test: very serious error - server side')
```

Load the website, see the error triggered and recorded in Sentry

2) Change the error to
```
  throw new UserInputError('Test: less serious input error - client side')
```

and see that the error is not recorded in Sentry


# Checklist (Definition of Done)
